src/Compilers/CSharp/Portable/Emitter/Model/NamedTypeSymbolAdapter.cs

```diff
@@ -668,6 +668,11 @@ IEnumerable<Cci.IMethodDefinition> Cci.ITypeDefinition.GetMethods(EmitContext co
             foreach (var method in this.GetMethodsToEmit())
             {
                 Debug.Assert((object)method != null);
+                if (method is SynthesizedStaticConstructor && context.Module.GetMethodBody(method) is null)
```

Comment:
>is SynthesizedStaticConstructor [](start = 27, length = 31)

I searched for ```is SynthesizedStaticConstructor``` within the project and couldn't find any match. However, it is unlikely that we don't already do an equivalent check somewhere. I think that kind of check would be preferred over a type check.

Reply:
See for example MethodSymbol.IsImplicitConstructor or MethodSimbol.IsImplicitInstanceConstructor. I suggest adding similar helper next to them and use it.

---
In reply to: [410527156](https://github.com/dotnet/roslyn/pull/43234#discussion_r410527156) [](ancestors = 410527156)